### PR TITLE
Filename pattern fixes

### LINKED ISSUE #609: Substitution parameter not supported in Image File Sink and Video File Sink
The documentation for Image File Sink and Video File Sink says:

>` DIR_LOCATION`: a location to write files to; can be a regular path or a path template; supported substitution parameters are `%source_id` and `%src_filename`;

But only JSON Metadata Sink supports this. In the code, I don't see any pattern replacement being done for Image File Sink and Video File Sink.

### LINKED ISSUE #772: %source_id and %src_filename are not respected in image_file sink adapter
Running on 0.2.10, the two params are not respected when used in the `DIR_LOCATION` environment variable. It instead will output to a path like: `/some/path/%source_id-%src_filename`

### PR DESCRIPTION
**Pull request by bwsw**
Closes #772 
Closes #609

### CODE REVIEW

File: adapters/python/sinks/metadata_json.py [hunk 1]
```diff
@@ -34,8 +34,9 @@ class Patterns:
 
 class MetadataJsonWriter(ChunkWriter):
     def __init__(self, pattern: str, chunk_size: int):
-        self.pattern = pattern
         super().__init__(chunk_size, logger_prefix=LOGGER_NAME)
+        self.pattern = pattern
+        self.logger.info(f'File name pattern is {self.pattern}')
```
Comment (tomskikh) [inline adapters/python/sinks/metadata_json.py]:
```suggestion
        self.logger.info('File name pattern is %s', self.pattern)
```

File: adapters/python/sinks/metadata_json.py [hunk 2]
```diff
@@ -60,7 +61,7 @@ def _write_meta_to_file(
         if frame_num is not None:
             metadata['frame_num'] = frame_num
         try:
-            json.dump(metadata, self.file)
+            json.dump(metadata, self.file, indent=4)
```
Comment (tomskikh) [inline adapters/python/sinks/metadata_json.py]: Make it optional to write pretty JSONs vs one JSON per line.

File: docs/source/savant_101/10_adapters.rst [hunk 3]
```diff
@@ -976,12 +1015,16 @@ The image file sink adapter extends the JSON metadata adapter by writing image f
 
 **Parameters**:
 
-- ``DIR_LOCATION``: a location to write files to; can be a regular path or a path template; supported substitution parameters are ``%source_id`` and ``%src_filename``;
+- ``DIR_LOCATION``: a location to write files to; can be a regular path or a path template; supported substitution parameters are ``%source_id``, ``%src_filename``, and ``%chunk_idx``;
```
Comment (tomskikh) [inline docs/source/savant_101/10_adapters.rst:1018]: Adding `%src_filename` does not work. It tooks a filename from the first frame. These are files, created by `image-json-sink` in `source_adapter_with_json_metadata` sample:
```
results/
├── coco-images-00000000-000000000785
│   ├── images
│   │   ├── 00000000.jpeg
│   │   ├── 00000001.jpeg
│   │   ├── 00000002.jpeg
│   │   ├── 00000003.jpeg
│   │   └── 00000004.jpeg
│   └── metadata.json
└── coco-images-00000001-000000000785
    ├── images
    │   ├── 00000000.jpeg
    │   ├── 00000001.jpeg
    │   ├── 00000002.jpeg
    │   ├── 00000003.jpeg
    │   └── 00000004.jpeg
    └── metadata.json
```

Sink parameters:
```
      - CHUNK_SIZE=5
      - DIR_LOCATION=/results/%source_id-%chunk_idx-%src_filename
```

File: adapters/python/sinks/image_files.py [hunk 4]
```diff
@@ -113,16 +112,19 @@ def _write_video_frame(self, video_frame: VideoFrame, content: bytes) -> bool:
             return False
         writer = self.writers.get(video_frame.source_id)
         if writer is None:
-            base_location = os.path.join(self.location, video_frame.source_id)
-            if self.chunk_size > 0:
-                json_filename_pattern = f'{Patterns.CHUNK_IDX}.json'
-            else:
-                json_filename_pattern = 'meta.json'
+            src_file_location = get_tag_location(video_frame) or ''
+            base_location = get_location(
```
Comment (tomskikh) [inline adapters/python/sinks/image_files.py]: Use `base_location` as a key for writer, like in metadata_json.py. That should fix the problem with `%src_filename`.
Same for video_files.py.

**Review by tomskikh — CHANGES_REQUESTED** [review]

File: adapters/python/sinks/metadata_json.py (hunk 2 above)
Comment (bwsw) [reply adapters/python/sinks/metadata_json.py]: I removed the identation as I do not want to introduce new parameters.

**Review by tomskikh — APPROVED** [review]

**Pull request merged**